### PR TITLE
Convert sanitize_allow to fastAPI

### PR DESCRIPTION
**Pull request by assuntad23**
Converting sanitize_allow to fastAPI, as per #10889

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [x] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

File: lib/galaxy/webapps/galaxy/api/sanitize_allow.py [hunk 1]
```diff
@@ -35,9 +38,8 @@ def create(self, trans, tool_id, **kwd):
             self._save_allowlist(trans)
         return self._generate_allowlist(trans)
 
-    @web.require_admin
-    @web.expose_api
-    def delete(self, trans, tool_id, **kwd):
+    @router.delete("/api/sanitize_allow", require_admin=True)
+    async def delete(self, tool_id, trans: ProvidesUserContext = DependsOnTrans):
```
Comment (mvdbeek) [inline lib/galaxy/webapps/galaxy/api/sanitize_allow.py]:
```suggestion
   def delete(self, tool_id, trans: ProvidesUserContext = DependsOnTrans):
```

File: lib/galaxy/webapps/galaxy/api/sanitize_allow.py [hunk 2]
```diff
@@ -1,31 +1,34 @@
...
         GET /api/sanitize_allow
         Return an object showing the current state of the toolbox and allow list.
         """
         return self._generate_allowlist(trans)
 
-    @web.require_admin
-    @web.expose_api
-    def create(self, trans, tool_id, **kwd):
+    @router.put("/api/sanitize_allow", require_admin=True)
+    async def create(self, tool_id, trans: ProvidesUserContext = DependsOnTrans):
```
Comment (mvdbeek) [inline lib/galaxy/webapps/galaxy/api/sanitize_allow.py]:
```suggestion
   def create(self, tool_id, trans: ProvidesUserContext = DependsOnTrans):
```

File: lib/galaxy/webapps/galaxy/api/sanitize_allow.py [hunk 3]
```diff
@@ -1,31 +1,34 @@
...
-class SanitizeAllowController(BaseAPIController):
-    @web.require_admin
-    @web.expose_api
-    def index(self, trans, **kwd):
+@router.cbv
+class FastAPISanitizeAllowController:
+    @router.get("/api/sanitize_allow", require_admin=True)
+    async def index(self, trans: ProvidesUserContext = DependsOnTrans):
         """
         GET /api/sanitize_allow
```
Comment (mvdbeek) [inline lib/galaxy/webapps/galaxy/api/sanitize_allow.py]:
```suggestion
```

File: lib/galaxy/webapps/galaxy/api/sanitize_allow.py [hunk 4]
```diff
@@ -35,9 +38,8 @@ def create(self, trans, tool_id, **kwd):
             self._save_allowlist(trans)
         return self._generate_allowlist(trans)
 
-    @web.require_admin
-    @web.expose_api
-    def delete(self, trans, tool_id, **kwd):
+    @router.delete("/api/sanitize_allow", require_admin=True)
+    async def delete(self, tool_id, trans: ProvidesUserContext = DependsOnTrans):
         """
         DELETE /api/sanitize_allow
```
Comment (mvdbeek) [inline lib/galaxy/webapps/galaxy/api/sanitize_allow.py]:
```suggestion
```

**Review by mvdbeek — COMMENTED** [review]
That's a nice start, can you also describe the path parameters and setup response models ?

File: lib/galaxy/webapps/galaxy/api/sanitize_allow.py [hunk 5]
```diff
@@ -1,31 +1,34 @@
...
 
-from galaxy import web
-from galaxy.webapps.base.controller import BaseAPIController
+from galaxy.managers.context import ProvidesUserContext
+from galaxy.webapps.galaxy.api import DependsOnTrans
+from . import Router
 
 log = logging.getLogger(__name__)
 
+router = Router(tags=["sanitize_allow"])
```
Comment (mvdbeek) [inline lib/galaxy/webapps/galaxy/api/sanitize_allow.py]:
```suggestion
router = Router(tags=["configuration"])
```

File: lib/galaxy/webapps/galaxy/api/sanitize_allow.py [hunk 6]
```diff
@@ -1,31 +1,34 @@
...
         """
         return self._generate_allowlist(trans)
 
-    @web.require_admin
-    @web.expose_api
-    def create(self, trans, tool_id, **kwd):
+    @router.put("/api/sanitize_allow", require_admin=True)
+    async def create(self, tool_id, trans: ProvidesUserContext = DependsOnTrans):
         """
         PUT /api/sanitize_allow
```
Comment (mvdbeek) [inline lib/galaxy/webapps/galaxy/api/sanitize_allow.py]:
```suggestion
```

**Comment by mvdbeek** [comment]
@assuntad23 you can batch the addition of suggestions from the review tab

File: lib/galaxy/webapps/galaxy/api/sanitize_allow.py (hunk 5 above)
Comment (assuntad23) [reply lib/galaxy/webapps/galaxy/api/sanitize_allow.py]: @mvdbeek not sure why I should do this? shouldn't the router tags match the endpoint?

Comment (mvdbeek) [reply lib/galaxy/webapps/galaxy/api/sanitize_allow.py]: there's no need for this to be a top level entry in the documentation (that's what tags are for), you're just changing configuration values here.

Comment (mvdbeek) [reply lib/galaxy/webapps/galaxy/api/sanitize_allow.py]: and this shouldn't have been a top level route, but now we got the mess, let's at least improve the documentation.

**Comment by assuntad23** [comment]
> That's a nice start, can you also describe the path parameters and setup response models ?

I took a shot at it; let me know how to improve.

**Comment by bgruening** [comment]
@assuntad23 I'm cleaning up the original issues. Are you working on this one still?

Happy new year!

**Comment by assuntad23** [comment]
@bgruening I beleive this was a swing and a miss. It is not suitable to be merged, as I was not entirely on the right page for how to format the routes. Let's close this for now, and perhaps in the future I'll take another stab at it, or, in the meantime, someone else can have a go.

**Pull request closed without merging**